offline_verifying.md

```diff
@@ -0,0 +1,108 @@
+# Offline Verifier
+
+## Problem
+
+Currently, after a document has been issued in the Document Creator, the user has to download the file before clicking the Verify Documents tab and uploading the document for verification which may be cumbersome. Created documents can be easily accessed through their `links.self.href` attribute only because the Document creator uses this function to upload the document to a server:
+
+```js
+export const uploadToStorage = async (
+  doc: WrappedDocument,
+  documentStorage: DocumentStorage
+): Promise<AxiosResponse> => {
+  const qrCodeObj = decodeQrCode(doc.rawDocument.links.self.href);
+  const uri = qrCodeObj.payload.uri;
+
+  return axios({
+    method: "post",
+    url: uri,
+    headers: getHeaders(documentStorage),
+    data: {
+      document: doc.wrappedDocument,
+    },
+  });
+};
+```
+However, as the user may not have consented to an upload, this poses a problem in our current workflow. The current implementation of easy verification of a document can be found in the [gallery](https://gallery.openattestation.com/).
+
+## Goal
+
+To explore the option of opening a document without uploading it to a server. This will save the step of downloading, opening another window and uploading the document. Possible usecases include: Allowing a user to view a .tt document with .tt files as attachments easily in another tab. 
+
+## Proposed Solution
+
+We can use the postMessage method to conduct parent and child communication. To detect the message sent, an eventListener can be added in the DropZone. For example:
+
+```js
+  window.addEventListener(
+    "message",
+    (event) => {
+      if (event.data.document) {
+        updateCertificate(event.data.document);
+      }
+    },
+    false
+  );
+
+```
+
+In order for the parent to communicate with its child window, a reference can be saved to the return value of `window.open()`. For example:
+
+```js
+  const childWin = window.open(url, "_blank");
+```
+
+Once the reference is made, we will establish a 3-way handshake to ensure that messages can be sent from parent to child and child to parent. This is to 
+ensure that a connection is established and both parties synchronize (SYN) and acknowledge (ACK) each other. Here is a simple illustration of how it is done:
+
+![3-way handshake](./assets/offline_verifier/tcp.png)
+
+When the child window is ready, it will send a SYN message to the parent through:
+
+```js
+window.opener.postMessage({ messageType: "SYN" });
+```
+
+When the parent receives the SYN message, he will send out a SYNACK message to its child:
+
+```js
+if (event.data.messageType == "SYN") {
+  childWin.postMessage({ messageType: "SYNACK" });
+}
+```
+
+When the child receives the SYNACK message, he will send out an ACK message back to its parent:
+
+```js
+if (event.data.messageType == "SYNACK") {
+  window.opener.postMessage({ messageType: "ACK" });
+}
+```
```

Comment:
Understand that this is borrowed from the TCP/IP handshake, I then ask myself the question what was the value of a 3 way handshare. 

Turns out the value of a 3-way handshake is to establish the initial sequence number (ISN) which is random in TCP/IP situation. And that a 2 way handshake will only guarantee the ISN reciept of one party. See discussion here: https://networkengineering.stackexchange.com/questions/24068/why-do-we-need-a-3-way-handshake-why-not-just-2-way. 

Comparing it to this situation I dont see a 3 way handshake being necessary because there is no need for a sequence number for packet re-assembly at the recipient side, and that we dont even break our message in multiple separate packets.

So, I think we can drop the 3 way handshake in favor of a more direct and simpler architecture where the sequence will look like:

1. Child sends "READY"
2. Parent sends "LOAD_DOCUMENT" with payload